pkg/adapter/v2/util/crstatusevent/eventsstatus.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crstatusevent
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+	"k8s.io/apimachinery/pkg/runtime"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+)
+
+type crStatusEvent struct {
+	cubeClient    *kubernetes.Clientset
+	Recorder      record.EventRecorder
+	Logf          func(format string, args ...interface{})
+	source        runtime.Object
+	component     string
+	kubeEventSink *typedcorev1.EventSinkImpl
+}
+
+var contextkey struct{}
+
+func ContextWithCRStatus(ctx context.Context, kubeEventSink *typedcorev1.EventSinkImpl, component string, source runtime.Object, logf func(format string, args ...interface{})) context.Context {
+
+	return context.WithValue(ctx, contextkey, &crStatusEvent{
+		component:     component,
+		Logf:          logf,
+		kubeEventSink: kubeEventSink,
+		source:        source.DeepCopyObject(),
+	})
+}
+
+func CRStatusFromContext(ctx context.Context) (cs *crStatusEvent, ok bool) {
+
+	cs, ok = ctx.Value(contextkey).(*crStatusEvent)
+
+	return
+}
+
+func ReportCRStatusEvent(ctx context.Context, result protocol.Result) {
+
+	var fromContext *crStatusEvent
+	ok := false
+	if fromContext, ok = CRStatusFromContext(ctx); !ok {
```

Comment:
maybe instead of `!ok`, do `ok` so that you don't have to declare the variable. Or do this: https://play.golang.org/p/uRfCxR9bLSP